raster-foundry/raster-foundry
# Void cache when layer mask changes

## Overview

This PR includes `mask` in the `BacksplashImage` construction params so that when it changes, the cache key for the image changes.

### Checklist


### Notes

Opted against doing the masking in the BacksplashImages themselves since that would mean _all_ of them would need to evaluate the mask, when sounds worse than masking once at the end. I don't have any evidence that it actually is though, and the cost is that it looks like BacksplashImage is carrying around a useless param.

## Testing Instructions

- turn your cache all the way on
- visualize a project layer in the v2 UI
- add an AOI if it doesn't have one
- the visualization should respect your mask

![](https://media1.tenor.com/images/2c08c9b6c2c35b7074b98f4e79cad26e/tenor.gif?itemid=6037351)

- make your mask weirder
- confirm
- the visualization should respect your new mask

![](https://media2.giphy.com/media/HCTfYH2Xk5yw/giphy.gif)

- make it weirder
- confirm
- the visualization should respect your new mask

![](https://media3.giphy.com/media/NnyqfcowpXZOU/giphy.gif)

- make :clap: it :clap: weirder
- :clap: confirm :clap:
- the :clap: visualization :clap: should :clap: respect :clap: your :clap: new :clap: mask

Closes #4850
